include/qfm/pricing/pricing.hpp

```diff
@@ -19,7 +19,7 @@ namespace pricing {
 class Pricing {
  public:
   void SetModel(std::shared_ptr<model::Model> model,
-                const asset::AssetType type) noexcept;
+                asset::AssetType type) noexcept;
```

Comment:
Why did you remove `const` here?

Reply:
That's a good question, I asked myself the same when I was looking at the changes. The warning given by `clang-tidy` is the following

```txt
/Volumes/workplace/quant_finance_models/include/qfm/pricing/pricing.hpp:22:17: error: parameter 'type' is const-qualified in the function declaration; const-qualification of parameters only has an effect in function definitions [readability-avoid-const-params-in-decls,-warnings-as-errors]
   22 |                 const asset::AssetType type) noexcept;
      |                 ^~~~~
```

The reason is (I believe) that `type` is passed by value (it's copied), hence marking it as `const` in the **header** (`.hpp`) file adds not value to the consumer (is a implementation detail). In this case seems that best practice is to remove unnecessary information to the consumer, and maintain those implementation details only the `.cpp` file (you can check that the `const` is still there). This is the case when passing parameters by value, not by reference.

btw, this removal (and quite a few others) was automatically done by `clang-tidy` with `--fix` tag.

```prompt
$ find src include main.cpp -name '*.cpp' -o -name '*.hpp' | xargs -n1 clang-tidy --fix 
```

Reply:
What about making it a constant reference instead? No reason to copy, I think.

Reply:
Yeah, that makes sense to me. Done!